lte/gateway/c/core/oai/tasks/amf/amf_ue_context_storage.cpp

```diff
@@ -0,0 +1,170 @@
+    return NULL;
+  }
+
+  new_p->amf_ue_ngap_id  = generate_amf_ue_ngap_id();
+  new_p->gnb_ngap_id_key = INVALID_GNB_UE_NGAP_ID_KEY;
+  new_p->gnb_ue_ngap_id  = INVALID_GNB_UE_NGAP_ID;
+
+  // Initialize timers to INVALID IDs
+  new_p->m5_mobile_reachability_timer.id    = AMF_APP_TIMER_INACTIVE_ID;
+  new_p->m5_implicit_detach_timer.id        = AMF_APP_TIMER_INACTIVE_ID;
+  new_p->m5_initial_context_setup_rsp_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_INITIAL_CONTEXT_SETUP_RSP_TIMER_VALUE};
+  new_p->m5_ulr_response_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_ULR_RESPONSE_TIMER_VALUE};
+  new_p->m5_ue_context_modification_timer = (amf_app_timer_t){
+      AMF_APP_TIMER_INACTIVE_ID, AMF_APP_UE_CONTEXT_MODIFICATION_TIMER_VALUE};
+  new_p->mm_state = DEREGISTERED;
+
+  new_p->amf_context._security.eksi = KSI_NO_KEY_AVAILABLE;
+  new_p->mm_state                   = DEREGISTERED;
+  new_p->amf_context.ue_context     = new_p;
+
+  return new_p;
+}
+
+// id-AMF-UE-NGAP-ID <--> ue_m5gmm_context_t map
+bool AmfUeContextStorage::amf_insert_into_amfid_ue_context_map(
+    std::shared_ptr<ue_m5gmm_context_t> pContext) {
+  if (!pContext && (INVALID_AMF_UE_NGAP_ID == pContext->amf_ue_ngap_id)) {
+    return false;
+  }
+  amfid_ue_context_map.insert_or_update(pContext->amf_ue_ngap_id, pContext);
+  return true;
+}
+bool AmfUeContextStorage::amf_remove_from_amfid_ue_context_map(
+    amf_ue_ngap_id_t ue_amf_id) {
+  magma::map_rc_t status = amfid_ue_context_map.remove(ue_amf_id);
+
+  if (magma::MAP_OK != status) {
+    return false;
+  }
+  return true;
+}
+
+std::shared_ptr<ue_m5gmm_context_t>
+AmfUeContextStorage::amf_get_from_amfid_ue_context_map(
+    amf_ue_ngap_id_t ue_amf_id) {
+  std::shared_ptr<ue_m5gmm_context_t> ctx;
+  magma::map_rc_t status = amfid_ue_context_map.get(ue_amf_id, ctx);
+  return ctx;
+}
+
+// id-GNB-UE-NGAP-ID <---> ue_m5gmm_context_t map
+bool AmfUeContextStorage::amf_insert_into_gnbid_ue_context_map(
+    std::shared_ptr<ue_m5gmm_context_t> pContext) {
+  if (!pContext && (INVALID_GNB_UE_NGAP_ID_KEY == pContext->gnb_ngap_id_key)) {
+    return false;
+  }
+  gnbid_ue_context_map.insert_or_update(pContext->gnb_ngap_id_key, pContext);
+  return true;
+}
+bool AmfUeContextStorage::amf_remove_from_gnbid_ue_context_map(
+    gnb_ue_ngap_id_t ue_gnb_id) {
+  magma::map_rc_t status = gnbid_ue_context_map.remove(ue_gnb_id);
+
+  if (magma::MAP_OK != status) {
+    return false;
+  }
+  return true;
+}
+std::shared_ptr<ue_m5gmm_context_t>
+AmfUeContextStorage::amf_get_from_gnbid_ue_context_map(
+    gnb_ue_ngap_id_t ue_gnb_id) {
+  std::shared_ptr<ue_m5gmm_context_t> ctx;
+  magma::map_rc_t status = gnbid_ue_context_map.get(ue_gnb_id, ctx);
+  return ctx;
+}
+
+// GUTI <---> ue_m5gmm_context_t map
+bool AmfUeContextStorage::amf_insert_into_guti_ue_context_map(
+    std::shared_ptr<ue_m5gmm_context_t> pContext) {
```

Comment:
let's not mix variable naming convention here, let's stick to what we are using through OAI as of now, which is underscore separated variables

Reply:
addressed review comment